src/test/java/uk/gov/ons/ctp/response/casesvc/message/TestPubSubMessage.java

```diff
@@ -32,14 +33,36 @@ public CaseNotificationDTO getPubSubCaseNotification() throws IOException {
           consumer.ack();
         };
     Subscriber subscriber = pubsubEmulator.getEmulatorSubscriber(receiver);
-    Thread t =
-        new Thread() {
+    Thread t = new Thread(() -> subscriber.startAsync().awaitRunning());
+    ExecutorService service = Executors.newSingleThreadExecutor();
+    service.execute(t);
+    while (receivedMessage == null) {
+      try {
+        Thread.sleep(10);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
+    subscriber.stopAsync().awaitTerminated();
+    service.shutdown();
+    System.out.println(receivedMessage);
+    ObjectMapper objectMapper = new ObjectMapper();
+    return objectMapper.readValue(receivedMessage, CaseNotificationDTO.class);
+  }
+
+  public CaseActionEvent getPubSubCaseActionEventStatus() throws IOException {
 
-          @Override
-          public void run() {
-            subscriber.startAsync().awaitRunning();
-          }
+    MessageReceiver receiver =
+        (PubsubMessage message, AckReplyConsumer consumer) -> {
+          // Handle incoming message, then ack the received message.
+          log.info("Id: " + message.getMessageId());
+          log.info("Data: " + message.getData().toStringUtf8());
+          receivedMessage = message.getData().toStringUtf8();
+          log.info("Received : " + receivedMessage);
+          consumer.ack();
         };
+    Subscriber subscriber = pubsubEmulator.getEmulatorSubscriberForCaseActionEventStatus(receiver);
+    Thread t = new Thread(() -> subscriber.startAsync().awaitRunning());
     ExecutorService service = Executors.newSingleThreadExecutor();
     service.execute(t);
     while (receivedMessage == null) {
```

Comment:
as above

Reply:
yes it can be null. So basically its a thread based approach where in the executer keeps checking if the emulator has received the message on the subscription to be picked up.